test/server/cards/02_SHD/leaders/HunterOutcastSergeant.spec.ts

```diff
@@ -0,0 +1,75 @@
+describe('Hunter, Outcast Sergeant', function () {
+    integration(function (contextRef) {
+        describe('Hunter\'s leader undeployed ability', function () {
+            it('should reveal a resource and bring back to hand if it share a name with a friendly unit in play and resource the top card', function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['battlefield-marine', 'echo#restored'],
+                        spaceArena: ['millennium-falcon#piece-of-junk'],
+                        resources: ['millennium-falcon#landos-pride', 'echo#restored', 'battlefield-marine', 'devotion'],
+                        leader: 'hunter#outcast-sergeant',
+                    },
+                    player2: {
+                        resources: ['echo#restored']
```

Comment:
Can we add a unique card that is inthe opponent's arena as well to confirm it's not included in the targeting? Could just give player 2 like a deployed Leia leader and put the Leia unit in resources for player 1